File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/Edit_blog_article_022_Vulnerable.py
articles = {
    1: {
        "title": "First Article",
        "content": "Original content"
    },
    2: {
        "title": "Second Article",
        "content": "Another content"
    }
}


def edit_blog_article(article_id, new_title, new_content):
    """
    Edits an existing blog article.
    """

    if article_id not in articles:
        return "Article not found."

    articles[article_id]["title"] = new_title
    articles[article_id]["content"] = new_content

    return "Article updated successfully."


if __name__ == "__main__":
    print(edit_blog_article(
        1,
        "Updated Title",
        "Updated article content"
    ))

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/Edit_blog_article_022_Vulnerable.java
import java.util.HashMap;
import java.util.Map;

public class Edit_blog_article_022_detelid {

    static Map<Integer, String[]> articles = new HashMap<>();

    static {
        articles.put(1, new String[]{
                "Old Title",
                "Old Content"
        });
    }

    public static String editArticle(
            int articleId,
            String title,
            String content) {

        if (!articles.containsKey(articleId))
            return "Article ID not found.";

        articles.put(articleId,
                new String[]{title, content});

        return "Article updated successfully.";
    }

    public static void main(String[] args) {

        System.out.println(editArticle(
                1,
                "New Title",
                "Updated Content"
        ));
    }
}